fix: remove grade assignments and penalty; add planned enhancements (#267)
AY2425S2-CS2103T-T12-1/tp: docs/DeveloperGuide.md

## **Appendix: Planned Enhancements**

Team size: 5

### Submission and grading of assignments

> Currently we have a way to add, delete and edit assignments, but no support for submission and grading of assignments, making the assignments feature rather useless at this point. Hence we have a planned enhancement to support submission and grading of assignments so that the assignment features make sense. This includes submission deadlines, which will add a specific submission time as a deadline rather than the current date deadline. The submission feature will allow TAs to record down the time that students submitted their assignments (which could be automated), and the grading feature will allow TAs to grade the students work, recording the grades down in the application.

Reviewer: We can make this sound nicer on ourselves by removing the deprecating parts and rephrasing others.

```suggestion
Currently, the app allows TAs to add, delete and edit assignments, but there is no support for submission and grading of assignments. We plan to allow TAs to add and grade student submissions for assignments. TAs will also be able to add submission deadlines and set penalties for late submissions. Ultimately, the goal is for TAs to be able to create assignment deadlines, record down their students' time of submission, and grade students' work while also penalizing late submissions.
```

### Display of assignment data
Currently assignments cannot be viewed as there is not much to be done with them without a grading feature. We plan to make them displayable together with the implementation of grading of assignments in the future so that the information is meaningful. This will be done via some sort of `list-assignments` command that will show what assignments are in the group, and for each student if they have submitted or not.

### Nicer UI and output of command feedback
Currently shows JSON output which is informative and clear for a technically inclined person, but does not look as nice on the UI. Future plan is to enhance the output format to be more visually appealing alongside a UI visual upgrade to display both person and group lists side-by-side for better information tracking. We also intend to make the feedback window dynamically sized so that error messages fit within the box and users will not need to scroll through, thus reducing friction when inputting wrong commands.

> ### Support for TA and Professor Roles

Reviewer: Capitalization
```suggestion
### Support for TA and Lecturer roles
```

Currently all persons in the application are assumed to be students which is most important to track for a TA. We plan to add support for TA and Professor roles which would facilitate the feature on grading and submission of assignments. 

> ### Listing Students in Sorted Name Order

Reviewer: Capitalization
```suggestion
### Listing students in sorted name order
```

Currently students in groups are indexed arbitrarily. For a future enhancement, we intend to list them in sorted name order for easier lookup.

> ### Marking attendance of Multiple students at once

Reviewer: Capitalization and adding support for marking multiple weeks of attendance

```suggestion
### Marking the attendance of multiple students for multiple weeks at once
```

> Right now our attendance marking is done one student at a time, which can be time consuming for large classes. We intend to allow the function to mark multiple students' attendance at once for a more efficient experience. The command would be something like `mark-all-attendance g/GROUP w/WEEK n/PERSON1 [n/PERSON2 n/PERSON3 ...]`.

Reviewer: Maybe we can also add that we intend to allow multiple weeks of attendance to be marked simultaneously.

```suggestion
Currently, `mark-attendance` only allows the attendance of a single student and for a single week to be marked. This can be time-consuming when marking attendance for larger classes. To make the experience more efficient, we plan to allow TAs to mark multiple students' attendance for multiple weeks at once. The command would look like `mark-all-attendance g/GROUP w/WEEK1[,WEEK2,WEEK3,...] n/PERSON1 [n/PERSON2 n/PERSON3 ...]`.
```

Reviewer: On second thought, a more consistent format for specifying weeks would be `w/WEEK1 [w/WEEK2 w/WEEK3 ...]`
